exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerBuilder.java

```diff
@@ -150,6 +151,11 @@ public PrometheusHttpServerBuilder setDefaultAggregationSelector(
    * registered with a {@link io.opentelemetry.sdk.metrics.SdkMeterProvider}.
    */
   public PrometheusHttpServer build() {
+    if (memoryMode == MemoryMode.REUSABLE_DATA && executor != null) {
+      throw new IllegalArgumentException(
+          "MemoryMode is REUSEABLE_DATA cannot be used with custom executor, "
+              + "since data may be corrupted if reading metrics concurrently");
```

Comment:
Is this documented anywhere else?

Reply:
we'll need to update the [docs](https://opentelemetry.io/docs/languages/java/configuration/) and change `otel.java.experimental.exporter.memory_mode` to `otel.java.exporter.memory_mode`, so we could potentially add in another footnote when we do that.